doc/nrf/app_dev/device_guides/hpf/rt_peripherals.rst

```diff
@@ -0,0 +1,320 @@
+  For more information on output shifting, see the *Shifting modes and usage* section in the VPR peripheral description.
+
+.. _hpf_real_time_input_shifting:
+
+Input shifting
+--------------
+
+VIO supports serialized input operations through shifting, activated by setting :c:enumerator:`NRF_VPR_CSR_VIO_MODE_IN_SHIFT`.
+Shifting process aligns with the serial clock and sampling point based on counter 1 event pulses.
+This mode is intended to be used in conjunction with :c:enumerator:`NRF_VPR_CSR_VIO_SHIFT_OUTB_TOGGLE` output mode.
+
+.. note::
+  For more information on input shifting, see the *Input Shifting* section in the VPR peripheral description.
+
+Combined access
+===============
+
+Since VIO is using 16 pins, some functionalities have been combined into a single register.
+By using ``nrf_vpr_csr_vio_*_combined_*``, you can decrease the number of times registers are accessed, and consequently improve timings.
+
+VPR Events Interface (VEVIF)
+****************************
+
+VEVIF is an event interface for VPR, allowing connection to the domain's DPPI system.
+VEVIF can also generate IRQs to other CPUs.
+Note that VEVIF is the only RT peripheral module with registers accessible through Advanced Peripheral Bus (APB).
+
+Tasks
+=====
+
+VEVIF tasks are responsible for generating interrupts handled by the VPR interrupt controller.
+You can trigger tasks in the following ways:
+
+* By calling the :c:func:`nrf_vpr_task_trigger` function, usually from other core to trigger an IRQ on VPR.
+* By calling the :c:func:`nrf_vpr_csr_tasks_set` function, which is done only locally from VPR to generate local IRQ.
+* Through the DPPI system, configured with the :c:func:`nrf_vpr_csr_vevif_subscribe_set` function.
+
+The interrupt service routine must clear the task bit using the :c:func:`nrf_vpr_csr_vevif_tasks_clear`.
+
+Events
+======
+
+VEVIF events can generate pulses that trigger an IRQ line or DPPI channels.
+You can trigger events through calling the :c:func:`nrf_vpr_csr_vevif_events_set` function (locally from VPR).
+To source DPPI using VEVIF events, you must configure it with the :c:func:`nrf_vpr_csr_vevif_publish_set` function.
+To source IRQ out of VPR using VEVIF events, you need to enable them using the :c:func:`nrf_vpr_int_enable` function (from remote side) or the :c:func:`nrf_vpr_csr_vevif_int_enable` function (from VPR).
+
+Buffering
+---------
+
+Similarly to VIO, you can buffer VEVIF events using the :c:func:`nrf_vpr_csr_vevif_events_buffered_set` function.
+Buffer overflows are avoided by stalling the CPU when writing to a dirty buffer.
+
+Tasks and events availability
+=============================
+
+The availability of VEVIF tasks and events varies depending on the SoC:
+
+.. list-table:: VEVIF availability on SoCs
+   :widths: auto
+   :header-rows: 1
+
+   * - Target
+     - VEVIF tasks available
+     - VEVIF tasks DPPI connections
+     - VEVIF events available
+     - VEVIF events DPPI connections
+
+   * - nRF54L15 FLPR
```

Comment:
tbc - Is it only nRF54L15? what about nRF54H20?

Reply:
HPF only targets L15.